src/log.ts

```diff
@@ -469,9 +569,16 @@ class Log {
    *   const apiResponse = data[0];
    * });
    */
-  info(entry, options, callback) {
+  info(entry: Entry|Entry[], options?: WriteOptions): Promise<ApiResponse>;
+  info(entry: Entry|Entry[], options: WriteOptions, callback: ApiResponseCallback): void;
+  info(entry: Entry|Entry[], callback: ApiResponseCallback): void;
+  info(entry: Entry|Entry[], optionsOrCallback?: WriteOptions|ApiResponseCallback, cb?: ApiResponseCallback):void|Promise<ApiResponse> {
+      const options =
+          typeof optionsOrCallback === 'object' ? optionsOrCallback : {};
+      const callback =
+          typeof optionsOrCallback === 'function' ? optionsOrCallback : cb;
       this.write(
-          Log.assignSeverityToEntries_(entry, 'INFO'), options, callback);
+          Log.assignSeverityToEntries_(entry, 'INFO'), options, callback!);
```

Comment:
Ditto.

It seems to me that there is a lot of code being duplicated here. Is there a way to avoid this?

Reply:
I tried :( I couldn't figure out a way to define the function signature once, and then effectively "implement" it with different names throughout.  Any ideas?

Reply:
I can't think of a way to do this that TypeScript is going to be happy with. Oh well.